Source/SwiftLintFramework/Rules/OperatorUsageWhitespaceRule.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  OperatorUsageWhitespaceRule.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 13/12/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+import SourceKittenFramework
+
+public struct OperatorUsageWhitespaceRule: OptInRule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "operator_usage_whitespace",
+        name: "Operator Usage Whitespace",
+        description: "Operators should be surrounded by a single whitespace " +
+                     "when they are being used.",
+        nonTriggeringExamples: [
+            "let foo = 1 + 2\n",
+            "let foo = 1 > 2\n",
+            "let foo = !false\n",
+            "let foo: Int?\n",
+            "let foo: Array<String>\n",
+            "let foo: [String]\n",
+            "let foo = 1 + \n  2\n",
+            "let range = 1...3\n",
+            "let range = 1 ... 3\n",
+            "let range = 1..<3\n"
+        ],
+        triggeringExamples: [
+            "let foo = 1+2\n",
+            "let foo = 1   + 2\n",
+            "let foo = 1   +    2\n",
+            "let foo = 1 +    2\n",
+            "let foo=1+2\n",
+            "let foo=1 + 2\n",
+            "let foo=bar\n",
+            "let range = 1 ..<  3\n"
+        ]
+    )
+
+    public func validateFile(_ file: File) -> [StyleViolation] {
+        let escapedOperators = ["/", "=", "-", "+", "*", "|", "^", "~"]
```

Comment:
I've removed `.`, `!` and `?` from the list to avoid false warnings. As they're less used, I don't see it as a large issue.